src/tests/functional/plugin/gpu/single_layer_tests/matmul.cpp

```diff
@@ -0,0 +1,378 @@
+
+            return std::make_shared<ngraph::Function>(results, params, "MatMul");
+        };
+        function = makeFunction(netType, params, matMul);
+    }
+};
+
+TEST_P(MatMulLayerGPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+}
+
+namespace {
+
+/* ============= Common params ============= */
+std::map<std::string, std::string> emptyAdditionalConfig;
+
+std::vector<std::map<std::string, std::string>> additionalConfig {
+    std::map<std::string, std::string>{/* empty config */},
+};
+
+const std::vector<ElementType> netPRCs {
+    ElementType::f32,
+};
+
+
+/* ============= FullyConnected ============= */
+namespace fullyConnected {
+
+const std::vector<ShapeRelatedParams> IS2D_smoke = {
+    {static_shapes_to_test_representation({{59, 1}, {1, 120}}), {false, true}},
+    {static_shapes_to_test_representation({{59, 1}, {1, 120}}), {true, true}},
+
+    {static_shapes_to_test_representation({{59, 120}, {120, 1}}), {false, false}},
+    {static_shapes_to_test_representation({{59, 120}, {120, 1}}), {true, true}},
+
+    {static_shapes_to_test_representation({{1, 120}, {120, 59}}), {false, false}},
+    {static_shapes_to_test_representation({{1, 120}, {120, 59}}), {true, false}},
+
+    {static_shapes_to_test_representation({{71, 128}, {128, 20}}), {true, false}},
+    {static_shapes_to_test_representation({{71, 128}, {128, 20}}), {false, true}},
+
+    {
+        {
+            {{-1, -1}, {{20, 60}, {20, 60}}},
+            {{60, 120}, {{60, 120}, {60, 120}}}
+        },
+        {false, false}
+    },
+    {
+        {
+            {{{0, 100}, {0, 12}}, {{20, 1}, {14, 1}, {20, 1}, {14, 1}}},
+            {{1, 120}, {{1, 120}, {1, 120}, {1, 120}, {1, 120}}}
+        },
+        {true, true}
+    }
+};
+
+const std::vector<ShapeRelatedParams> IS2D_nightly = {
+    {static_shapes_to_test_representation({{59, 1}, {1, 120}}), {false, false}},
+    {static_shapes_to_test_representation({{59, 1}, {1, 120}}), {true, false}},
+
+    {static_shapes_to_test_representation({{59, 120}, {120, 1}}), {true, false}},
+    {static_shapes_to_test_representation({{59, 120}, {120, 1}}), {false, true}},
+
+    {static_shapes_to_test_representation({{1, 120}, {120, 59}}), {true, true}},
+    {static_shapes_to_test_representation({{1, 120}, {120, 59}}), {false, true}},
+
+    {static_shapes_to_test_representation({{71, 128}, {128, 20}}), {true, true}},
+    {static_shapes_to_test_representation({{71, 128}, {128, 20}}), {false, false}},
+
+    {
+        {
+            {{-1, -1}, {{71, 128}, {50, 128}}},
+            {{128, 20}, {{128, 20}, {128, 20}}}
+        },
+        {false, false}
+    },
+    {
+        {
+            {{-1, 59}, {{10, 59}, {15, 59}, {15, 59}}},
+            {{59, 1}, {{59, 1}, {59, 1}, {59, 1}}}
+        },
+        {true, false}
+    },
+    {
+        {
+            {{{0, 120}, 59}, {{5, 59}, {11, 59}, {5, 59}, {10, 59}}},
+            {{59, 120}, {{59, 120}, {59, 120}, {59, 120}, {59, 120}}}
```

Comment:
Is there any reason that we should test only for the same output shape?

Reply:
In the test that you indicated the same second input only, it doesn't matter there. Output shape is different.
Maybe you mean the same input/output rank in 2D case?